addons/mail/models/mail_activity.py

```diff
@@ -376,14 +376,16 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
             right = DocumentModel.check_access_rights(doc_operation, raise_exception=False)
             if right:
                 valid_docs = DocumentModel.browse(doc_ids)._filter_access_rules(doc_operation)
+                valid_doc_ids = set(valid_docs.ids)
```

Comment:
Note that it could be inlined directly below

Reply:
By inlining it in the generator expression, won't the set be created at each loop?